dps310/dps310.go

```diff
@@ -0,0 +1,539 @@
+	// find temperatur src
+	var (
+		data uint8
+	)
+	data, err = d.readRegister(COEF_SRCE)
+	data &= COEF_SRCE_TMP_COEF_SR_MSK
+
+	// check rate oversamling combination
+	d.tempRate = rate
+	d.tempOs = os
+	if !d.isOkRateOs() {
+		return fmt.Errorf("invalid rate oversampling comination")
+	}
+
+	// add temp rate and oversampling
+	data += (uint8(rate)&TMP_CFG_TM_PRC_MSK)<<4 | (uint8(os) & TMP_CFG_TM_RATE_MSK)
+	err = d.writeRegister(TMP_CFG, data)
+	if err != nil {
+		return
+	}
+
+	data, err = d.readRegister(CFG_REG)
+	if err != nil {
+		return
+	}
+
+	if os > PRC_8SAMPLES {
+		err = d.writeRegister(CFG_REG, data| CFG_REG_TMP_SHIFT_EN_MSK)
+	} else {
+		err = d.writeRegister(CFG_REG, data& CFG_REG_TMP_SHIFT_DIS_MSK)
+	}
+	if err != nil {
+		return
+	}
+
+	d.tempScale = osScalingData[os].factor
+	return
+}
+
+// Read the temperatur
+func (d *DPS310) ReadTemperature() (temperature float32, err error) {
+
+	rawTemperature, err := d.readRawTemperature()
+	if err != nil {
+		return
+	}
+
+	// see: datasheet pages 14 and 15 
+	scaledRawtemp := rawTemperature / d.tempScale
+	temperature = scaledRawtemp*d.c1 + d.c0/2.0
+	return
+}
+
+// Read the values  from the sensor
+func (d *DPS310) ReadPressure() (pressure, temperature float32, err error) {
+
+	rawPressure, err := d.readRawPressure()
+	if err != nil {
+		return
+	}
+
+	rawTemperature, err := d.readRawTemperature()
+	if err != nil {
+		return
+	}
+
+	// see: datasheet pages 14 and 15 
+	scaledRawtemp := rawTemperature / d.tempScale
+	temperature = scaledRawtemp*d.c1 + d.c0/2.0
+
+	scaledRawpres := rawPressure / d.pressureScale
+	pressure =
+		d.c00 + scaledRawpres*(d.c10+
+			scaledRawpres*(d.c20+scaledRawpres*d.c30)) +
+			scaledRawtemp*d.c01 +
+			scaledRawpres*scaledRawtemp*(d.c11+scaledRawpres*d.c21)
+
+	return
+}
+
+// reads raw temperature
+func (d *DPS310) readRawTemperature() (temperature float32, err error) {
+	buf := d.rwBuf[:4]
+	buf[0] = 0x80 | TMP_B2
+	buf[1] = 0
+	buf[2] = 0
+	buf[3] = 0
+	err = d.tx(buf)
+	if err != nil {
+		return
+	}
+
+	temperature = decVal(uint24_t(buf[1])<<16 |
+		uint24_t(buf[2])<<8 |
+		uint24_t(buf[3]))
+	return
+}
+
+// reads raw pressure
+func (d *DPS310) readRawPressure() (pressure float32, err error) {
+	buf := d.rwBuf[:4]
+	buf[0] = SPIReadBit | PRS_B2
+	buf[1] = 0
+	buf[2] = 0
+	buf[3] = 0
+	err = d.tx(buf)
+	if err != nil {
+		return
+	}
+	pressure = decVal(uint24_t(buf[1])<<16 |
+		uint24_t(buf[2])<<8 |
+		uint24_t(buf[3]))
+	return
+}
+
+// Whether new temperature data is available
+// returns True if new data available to read
+func (d *DPS310) TemperatureAvailable() (bool, error) {
+	v, err := d.readRegister(MEAS_CFG)
+	return v&MEAS_CFG_TMP_RDY_MSK!= 0, err
+}
+
+// Whether new pressure data is available
+// returns True if new data available to read
+func (d *DPS310) PressureAvailable() (bool, error) {
+	v, err := d.readRegister(MEAS_CFG)
+	return v&MEAS_CFG_PRS_RDY_MSK != 0, err
+}
+
```

Comment:
A function `TemperatureAndPressureAvailable` would probably be useful, as both use the same register.

`waitDataAvailable` could then use this function.